src/analyzer.ts

```diff
@@ -15,8 +16,17 @@ export class Analyzer extends AnalyzerGen {
 
 	constructor(dartVMPath: string, analyzerPath: string) {
 		super();
+
 		console.log(`Starting Dart analysis server...`);
-		this.analyzerProcess = child_process.spawn(dartVMPath, [analyzerPath]);
+
+		let args = [analyzerPath];
+
+		// Optionally start the analyzer's diagnostic web server on the given port.
+		let port = getConfig('analyzerDiagnosticsPort');
+		if (port)
+			args.push(`--port=${port}`);
```

Comment:
What is this magic? I can't see any mention of a diagnostics port on the docs? !